File: 프로그래머스/2/81302. 거리두기 확인하기/거리두기 확인하기.java
public class Solution {
    public static int check(String[] place) {
        int[][] direction = {{0,1},{1,0},{0,-1},{-1,0}};
        for (int i = 0; i < 5; i++) {
            for (int j = 0; j < 5; j++) {
                int pcnt = 0;
                for (int[] d : direction) {
                    int dx = i + d[0];
                    int dy = j + d[1];
                    if (0 <= dx && dx < 5 && 0 <= dy && dy < 5 && place[dx].charAt(dy) == 'P') {
                        pcnt++;
                    }
                }
                if ((place[i].charAt(j) == 'P' && pcnt > 0) || (pcnt > 1 && place[i].charAt(j) == 'O')) {
                    return 0;
                }
            }
        }
        return 1;
    }

    public static int[] solution(String[][] places) {
        int[] answer = new int[places.length];
        for (int i = 0; i < places.length; i++) {
            answer[i] = check(places[i]);
        }
        return answer;
    }
}

File: 프로그래머스/2/81302. 거리두기 확인하기/거리두기 확인하기.py
def check(place):
    print()
    direction = [[0,1],[1,0],[0,-1],[-1,0]]
    for i in range(5):
        for j in range(5):
            pcnt = 0
            for a,b in direction:
                dx,dy = i+a,j+b
                if 0<=dx<5 and 0<=dy<5 and place[dx][dy]=='P':
                    pcnt+=1
            print([i,j],pcnt)
            if place[i][j]=='P' and pcnt or pcnt>1 and place[i][j]=='O':
                return 0
    return 1
            

def solution(places):
    answer = [check(place) for place in places]
    return answer